Formula/node.rb

```diff
@@ -56,30 +56,11 @@ def install
 
     if build.with? "npm"
       resource("npm").stage buildpath/"npm_install"
-
-      # make sure npm can find node
-      ENV.prepend_path "PATH", bin
```

Comment:
This is where I'm at now. I appreciate it's hideous, but so far in testing it is working. The error I chased around for like an hour last night seemed to be a red herring, so that was fun.

```diff
diff --git a/Library/Homebrew/language/node.rb b/Library/Homebrew/language/node.rb
index ab2206a5b..7137ce921 100644
--- a/Library/Homebrew/language/node.rb
+++ b/Library/Homebrew/language/node.rb
@@ -23,7 +23,17 @@ module Language
       # (or creating a new one if no .npmignore file already exists)
       open(".npmignore", "a") { |f| f.write("\n.brew_home\n") }
       # npm install args for global style module format installed into libexec
-      ["--verbose", "--global", "--prefix=#{libexec}", "."]
+      # npm 5.0.0 pretty much smashed the assumption Homebrew made that npm
+      # would always generate installable directories out-of-tree.
+      # We can restore the old behaviour that made package management
+      # relatively possible by packing the path up first. Obviously, this is
+      # not ideal.
+      %W[
+        --verbose
+        --global
+        --prefix=#{libexec}
+        #{Dir.pwd}/#{Utils.popen_read("npm pack").chomp}
+      ]
     end
 
     def self.local_npm_install_args
```